text/tls-and-alpn.rst

```diff
@@ -0,0 +1,380 @@
++-------------------+----------------------------+------------------------------------+------------------------------------+
+| ``edgeql+http``   | EdgeQL over HTTP           | ``/<database>?query=``             | ``/db/<database>/edgeql?query=``   |
++-------------------+----------------------------+------------------------------------+------------------------------------+
+| ``graphql``       | GraphQL over HTTP          | ``/<database>/explore``            | ``/db/<database>/graphql/explore`` |
++-------------------+----------------------------+------------------------------------+------------------------------------+
+| ``notebook``      | The Notebook extension     | ``/<database>/status``             | ``/db/<database>/notebook/status`` |
++-------------------+----------------------------+------------------------------------+------------------------------------+
+| <default>         | No ALPN protocol match     | ``/server/status/ready``           |                                    |
+|                   |                            |                                    |                                    |
+|                   |                            | ``/db/<database>/edgeql?query=``   |                                    |
+|                   |                            |                                    |                                    |
+|                   |                            | ``/db/<database>/graphql/explore`` |                                    |
+|                   |                            |                                    |                                    |
+|                   |                            | ``/db/<database>/notebook/status`` |                                    |
++-------------------+----------------------------+------------------------------------+------------------------------------+
+
+On the client side for either Option, the bindings or the CLI should provide an
+ALPN protocol based on the scenario. In most cases, the client binding should
+use ``edgedb-binary``
+
+
+Client Bindings & REPL
+----------------------
+
+The official client bindings for Python, Go and Node.js, as well as the EdgeDB
+CLI (the REPL and any command connecting to the server) will use TLS to talk to
+the EdgeDB instances by default. They should advertise the expected ALPN
+protocol in the TLS handshake, and be ready to handle any negotiation result,
+depending on the conclusion we'll make for previous questions.
+
+For compatibility considerations, the client is recommended to provide a
+temporary option to connect to old EdgeDB servers without TLS. However, this
+option should come with a warning and should not be default.
+
+For certificate verification, this was discussed previously. To recap, in order
+to make the client code simple and consistent, the client should either check
+the well-known EdgeDB location for the flag of self-signed certificate in order
```

Comment:
Let's have a well-known location of self-signed certificate itself, not to the flag, as described above. Then client don't have to hardcode a switch between verified and non-verified mode. It's always verified.

Specifically for CLI we can ask if we don't have a certificate yet, similarly to how SSH behaves on the first connect. But probably it's better to have certificate (or checksum of thereof) in the `credentials.json`. So that only file is enough to connect.